frontend/src/views/MainPage/MainPage.js

```diff
@@ -95,28 +95,33 @@ const MainPage = () => {
   };
 
   useEffect(() => {
+    const elements = document.querySelectorAll(
+      ".from-right, .from-left, .from-down, .from-bounce"
+    );
+
     const observer = new IntersectionObserver(entries => {
       entries.forEach(entry => {
         if (entry.isIntersecting) {
-          entry.target.classList.add("animate");
-          if (
-            entry.target.classList.contains("fade-in-left") ||
-            entry.target.classList.contains("fade-in-right")
-          ) {
-            setTimeout(() => {
-              entry.target.classList.add("delay-animate");
-            }, 1000); // 딜레이 시간을 1초로 설정
+          if (entry.target.classList.contains("from-right")) {
```

Comment:
const classMappings = {
  'from-right': 'fade-in-right',
  'from-left': 'fade-in-left',
  'from-down': 'fade-in-up',
  'from-bounce': 'fade-in-bounce'
};
function addFadeInClass(element) {
  for (const [key, value] of Object.entries(classMappings)) {
    if (element.classList.contains(key)) {
      element.classList.add(value);
      break; 
    }
  }
}